### ISSUE TITLE
Finish Essential Features

### ISSUE THREAD

**Issue by PatReynolds** [FreeUKGen/FreeCENMigration#743]
Essential features include
search by occupation
search by census location
search by birth  location

**Comment by Captainkirkdawson** [comment]
Before doing one line of code we need to have a clear understanding of the current nature of the search engine for FreeCEN2. 
It is currently designed to search for individuals specified by forename or surname or both born within a range of years who was either enumerated in a location or born in a location.
It is NOT designed to search for a) all males or females b) all disabled people c) all widowed people or many of the criteria that have been included in a census.
It is likely that CEN1 WOULD attempt such searches but time out or max record out. However we never told researchers what they could or could not do.
Currently all our second generation applications are **individual searches** as were REG1 and BMD1. 
Hence I propose that these criteria are applied as filters on a basic individual search using and extending the current search engine.
Fundamentally what that means is that the search will use the name/location/date/census year to locate individuals from the search_record collection. Those results would then be filtered by the additional criteria based on the content of the individual collection.  To assist in performance we would build indices on the individual collection as none currently exist.

**Comment by richpomfret** [comment]
@richpomfret to clarify the requirements with @PatReynolds and @FreecenBren.

**Comment by richpomfret** [comment]
We will otherwise go on the assumption that we search by people and then by x, y or z (e.g. occupation).

**Comment by Captainkirkdawson** [comment]
System of filters was developed as part of "the Christmas Present'. However it seems that is is likely to confuse people unless well explained.

**Comment by Captainkirkdawson** [comment]
@PatReynolds wrote

I think we have two options, keep the Christmas Presents on the search page, or put them on the search results page.  Suggested placement/text:

Option 1, I suggest the following layout:

Move soundex and the search button up - maybe even into the main box.  Nearby places will go here, in due course.
Delete "Please read the help carefully (etc)"

Below the search button, and below a line (or in a separate box) new text:

The following features will help you narrow down your search.  If you have run a search, use the 'amend search' button to return here.  Please read the help carefully (etc).

___________________________________

Option 2: amend this text, to include "Too many results? refine your search using one of the options [click here] or use the return button to narrow by year of birth, place of birth, or place of census."
![image (1).png](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/922ee9f6-b530-4c68-a3c0-7ca467984d0c)
At bottom of page (below table, above revise search, new search, etc. buttons, place "Christmas Present" options in a box with a button that says "filter current search" (or something).

The label of the 'revise search' button in this context will probably be confusing, so needs to be something like 'amend current search'..

What do you-all think?

**Comment by Captainkirkdawson** [comment]
@PatReynolds @richpomfret @DeniseColbert @AlOneill @Vino-S 
I moved Pat's emailed comment into this story so that we can retain the discussion for the future.
As an aside note that REG also uses filters for family members and witnesses not direct search fields.

Moving these filters to the Results page would be a massive undertaking. Thay are designed to be applied to the base results at the time of the search. Reprogramming is possible, would be a size 20 and likely take 2 months to deliver. So I do not like option 2

If it is desired that we treat these as truly search fields then we should re-engineer the search to use the individuals collection rather than the search records collection. All fields are there. I have no idea why CEN uses a search record collection other than that is what REG uses. Reprogramming would be a size 30 and likely take 4 months to deliver. So I do not like option 2

If we are to retain on the search page then it would make sense to move the soundex option up near the name fields just as done in BMD1 and proposed for 2. Also the nearby when it comes back goes near the place selection as there must be an initial place anyway. These two changes should likely be done ANYWAY regardless of anything else.

I dislike the idea of moving the search button below the set of filters. That button should always be the last item on any Search page

As to the text I have a preference for my own which refers to the help document; the warning about the system responding differently by census year is essential

The KEY in my mind is what we write into the Help material

I really would like to hear other opinions.

It seems clear we are not ready to release 'the Christmas Present' as part of this sprint release.

**Comment by PatReynolds** [comment]
Thanks, Kirk.  Agreed, not on the detailed results page - far too much work for too little gain.
Agred, moving the soundex up is a great idea.

I am obviously missing something, as I don't understand how witnesses and family members on FreeREG can be fiilters (e.g. if I search for with an Alcock, are you saying that the search actually finds all the Alcocks, but filters out everyone but the baptised, married or buried person (a sort of negative filter?).

Absolutely warnings about help, etc. must remain.

**Comment by Captainkirkdawson** [comment]
@PatReynolds  wrt
I am obviously missing something, as I don't understand how witnesses and family members on FreeREG can be filters (e.g. if I search for with an Alcock, are you saying that the search actually finds all the Alcocks, but filters out everyone but the baptised, married or buried person (a sort of negative filter?).
No we do not search for all the Alcocks and then filter out on the basis of "baptised, married or buried". 
We do search for Alcocks who are baptised (or buried) in the specified location and date range. We then apply a filter that uses the family member option setting. If the option is set we do nothing ie family members are there. If it is NOT set we filter out those results that are family members.
Similarly for marriages. We do a search for Alcocks who are married in the specified location and date range. We do nothing if family members and witnesses are selected. If they are not we filter out those not required eg if you do not want witnesses who are Alcocks we remove them.

This is how the embargo works. All records are retrieved and filter out those embargoed.

When do we use filters and when do we search? We search when we can use fields that are indexed. We filter when we need to use fields that are not indexed or are not in the search record collection at all

**Comment by PatReynolds** [comment]
Tahnks, Kirk, " If it is NOT set we filter out those results that are family members." is what I was trying to say.

**Comment by Captainkirkdawson** [comment]
That is correct. **If it is NOT set we filter out those results that are family members.**
However the original comment introduced **but filters out everyone but the baptised, married or buried person** and that is not the case baptised, married or buried are search fields not filters.

Filters ALWAYS work as removals never additions. The search gives us the full set we wish to examine. Works just like EXCEL filters

**Comment by Captainkirkdawson** [comment]
Have moved the soundex option to the Person options. Have renamed the Final Search box to Filters and changed wording in its chapeau.
Created story for writing help materials #795
Deployed to test3
